tools/dbootimg/src/dbootimg.c

```diff
@@ -430,14 +430,21 @@ static void *aboot_update_dtb(void *boot, void *dtb, bool force)
 static void *aboot_update_kernel(void *boot, void *kernel)
 {
 	struct aboot_hdr *aboot, *old_aboot = boot;
-	ssize_t page_sz, kernel_sz, align_sz;
+	ssize_t page_sz, kernel_sz, align_sz, dtb_sz;
 	void *old_kernel, *dtb, *ptr;
 	int diff;
 
 	page_sz = le32_to_cpu(old_aboot->page_size);
 	old_kernel = aboot_get_kernel(old_aboot);
 	kernel_sz = kernelgz_size(kernel);
-	dtb = aboot_get_dtb(old_aboot);
+
+	if (kernel_has_appended_dtb(kernel)){
```

Comment:
we don't really care if the new kernel has appended dtb, we just want to avoid error if the current one does not have dtb.
dtb = aboot_get_dtb(old_aboot);
dtb_sz = dtb ? dtb_size(dtb) : 0;